File: docs/snippets/all/archetypes/mcap_statistics_simple.cpp
// Log simple MCAP recording statistics.

#include <rerun.hpp>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_mcap_statistics");
    rec.spawn().exit_on_failure();

    rec.log(
        "mcap/statistics/recording_overview",
        rerun::archetypes::McapStatistics::update_fields()
            .with_message_count(12500)
            .with_schema_count(3)
            .with_channel_count(5)
            .with_attachment_count(2)
            .with_metadata_count(8)
            .with_chunk_count(25)
            .with_message_start_time(1743465600000000000) // 2024-04-01 00:00:00 UTC in nanoseconds
            .with_message_end_time(
                1743466200000000000 // 2024-04-01 00:10:00 UTC in nanoseconds (10 minute recording)
            )

    );
}

File: docs/snippets/all/archetypes/mcap_statistics_simple.rs
//! Log simple MCAP recording statistics.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_mcap_statistics").spawn()?;

    rec.log(
        "mcap/statistics/recording_overview",
        &rerun::McapStatistics::update_fields()
            .with_message_count(12500u64)
            .with_schema_count(3u64)
            .with_channel_count(5u64)
            .with_attachment_count(2u64)
            .with_metadata_count(8u64)
            .with_chunk_count(25u64)
            .with_message_start_time(1743465600000000000i64) // 2024-04-01 00:00:00 UTC in nanoseconds
            .with_message_end_time(1743466200000000000i64), // 2024-04-01 00:10:00 UTC in nanoseconds (10 minute recording)
    )?;

    Ok(())
}
